src/internal/cnd.c

```diff
@@ -202,12 +202,12 @@ r_obj* new_condition_names(r_obj* data) {
   return nms;
 }
 
-const char* rlang_obj_type_friendly_full(r_obj* x, bool value, bool length) {
-  r_obj* out_obj = KEEP(r_eval_with_xyz(obj_type_friendly_call,
-                                        x,
-                                        r_lgl(value),
-                                        r_lgl(length),
-                                        rlang_ns_env));
+// `length` is no longer a valid argument
+const char* rlang_obj_type_friendly_full(r_obj* x, bool value, bool _length) {
+  r_obj* out_obj = KEEP(r_eval_with_xy(obj_type_friendly_call,
```

Comment:
How do we manage deprecation here? I guess all downstream users should update, then at some point in the future we can remove this?

Reply:
We first need to add a new C callable, but I didn't bother here. This C callable is still valid. When we add a new one, we just forget this one for a few year and then remove it if we ever come across it again.